### PR TITLE
#114 nx refactor ui

### LINKED ISSUE #114: Refactor UI components as nx library
Refactor UI components as nx library. 

## Expected Behaviour

Move components from web app to component library:

- Atoms + molecules to be kept private.
- Organisms to be exposed.
- Web app to compose templates + pages from lib organisms.

## Current Behaviour

All components in web app.

## Possible Solution

See above.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #114.

Note: reviewing this might be quite impossible due to the amount of stuff moved around. Suggest just checking out the branch and making sure things work!

Some things to get you started (any tests that fail might be best fixed up in a follow-up, figured this PR was big enough as is!):

- `nx run ui:storybook`
- `nx run ui:test`
- `nx dep-graph` 😎

## Description

TLDR; starts refactor of react components into `nx` libraries!

Added following libraries:

- `ui`: core react component library consisting of material-ui components and generic custom components, should only be "dumb" components (i.e. no state management).
- `data`: type declarations and data access mappers (TODO).

Both UI libraries are wired up with `jest` and `storybook`, so as we use components we can add tests to the respective libraries (might be some discretion as to what should be a library test vs. app test, generally app tests for specific functionality, library for generic). 

Next steps: 

- extract scripts, config to `tools` (maybe config to root level `.env` ?)
- add scripts for generating unit + e2e test fixtures (e.g. https://medium.com/@wescopeland/how-to-build-custom-tools-in-a-nrwl-nx-workspace-e079e09ecffb).
- extract dgraph mappers etc. from legacy api to data library, e.g. mappers (see https://nx.dev/angular/workspace/structure/library-types).
- extract `web` graphql + state management (e.g. redux, sagas etc.) to data library (would be premature at this stage, as @mark-prins is still wiring all this up!).
- wire up the end-to-end tests for the `web` project with the Apollo backend (`nx` script would generate the fixtures, run the e2e tests). 

A lot to cover, but I think once it's all done will make it really easy to expand out the app as it grows (and will make for a really cool monorepo to show off)!

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [ ] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [ ] Tests OK.

### CODE REVIEW

File: unified-codes/apps/web/stories/EntityBrowser.stories.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
-  AlertProps,
-  EntityBrowser,
   Severity,
   Snackbar,
-} from "../src/components";
+} from "@unified-codes/ui";
+
+import {
+  EntityBrowser
+} from "@unified-codes/feature";
```
Comment (mark-prins) [inline unified-codes/apps/web/stories/EntityBrowser.stories.tsx:14]: this is the other `feature` import; which I think needs updating.

File: unified-codes/libs/data/.eslintrc [hunk 2]
```diff
@@ -0,0 +1 @@
+{ "extends": "../../.eslintrc", "rules": {}, "ignorePatterns": ["!**/*"] }
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/libs/data/.eslintrc:1]: Yeah, think it extends the root eslint and just gives the option to add some library specific things. All of this stuff was added by default by `nx` 🤷‍♂️.

File: unified-codes/.eslintrc [hunk 3]
```diff
@@ -10,11 +10,11 @@
   "ignorePatterns": ["**/*"],
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/.eslintrc:10]: Not sure! Automatically added by `nx`, so a bit scared to touch it!

File: unified-codes/libs/ui/src/lib/data/atoms/TableBody.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import Table from "./Table";
+import TableBody from "./TableBody";
+import TableCell from "./TableCell";
+import TableRow from "./TableRow";
+import TableHead from "./TableHead";
+
+describe(' TableBody', () => {
+  it('should render successfully', () => {
+    const { baseElement } = render(
+      <Table>
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/libs/ui/src/lib/data/atoms/TableBody.spec.tsx:13]: Yeah these were basically just stubs which I expanded out to pass. I think for material UI components we can probably remove tests (although retain storybook?).

File: unified-codes/apps/web/stories/EntityBrowser.stories.tsx (hunk 1 above)
Comment (wlthomson) [reply unified-codes/apps/web/stories/EntityBrowser.stories.tsx]: a8cd7e3ee1fec42148423fae1f81cebda196c312.

File: unified-codes/apps/web/src/components/templates/Explorer.tsx [hunk 5]
```diff
@@ -1,9 +1,9 @@
 import * as React from "react";
 import { useQuery, gql } from "@apollo/client";
 
-import { Grid } from "../atoms";
-import { EntityBrowser } from "../organisms";
-import { Entity, EntityNode } from "../../types";
+import { Grid } from "@unified-codes/ui";
+import { EntityBrowser } from "@unified-codes/feature";
```
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/web/src/components/templates/Explorer.tsx]: Oops! a8cd7e3ee1fec42148423fae1f81cebda196c312.

[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/web/src/components/templates/Explorer.tsx]: a8cd7e3ee1fec42148423fae1f81cebda196c312.

File: unified-codes/apps/web/stories/EntityBrowser.stories.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (wlthomson) [reply unified-codes/apps/web/stories/EntityBrowser.stories.tsx]: a8cd7e3ee1fec42148423fae1f81cebda196c312.
